habitat/tasks/nav/nav.py

```diff
@@ -453,9 +448,27 @@ def update_metric(
         ep_success = 0
         current_position = self._sim.get_agent_state().position.tolist()
 
-        distance_to_target = self._sim.geodesic_distance(
-            current_position, episode.goals[0].position
-        )
+        if self._config.DISTANCE_TO == "POINT":
+            distance_to_target = self._sim.geodesic_distance(
+                current_position, episode.goals[0].position
+            )
+        elif self._config.DISTANCE_TO == "VIEW_POINTS":
+            multi_goal = habitat_sim.MultiGoalShortestPath()
+            multi_goal.requested_start = current_position
+            multi_goal.requested_ends = self._episode_view_points
+            self._sim._sim.pathfinder.find_path(multi_goal)
+            distance_to_target = multi_goal.geodesic_distance
+
+        # remove after debug
+        # print(
+        #     f"distance_to_target: {distance_to_target}, delta d: {self._previous_distance_to_target - distance_to_target}"
+        # )
+        self._previous_distance_to_target = distance_to_target
+        if (
+            self._previous_distance_to_target - distance_to_target
+            > self._sim.config.FORWARD_STEP_SIZE
+        ):
+            print("!!! distance_to_target change more than forward step")
```

Comment:
This is possible with sliding on :) -- so these prints will just be a nuisance.

Reply:
@erikwijmans that was there for debugging purposes. With multi goal setup that was important that the geodesic distance metric is smooth when switching from one closest view point to another.

Reply:
Yeah, it'll be smooth.  The algorithm that is doing it in habitat-sim is doing `min(geo_dist(s, view_pt) for view_pt in view_points])`, just in a smart way.